src/main/java/org/folio/inventory/resources/Items.java

```diff
@@ -93,10 +95,15 @@ public void register(Router router) {
     router.put(RELATIVE_ITEMS_PATH_ID).handler(this::update);
     router.delete(RELATIVE_ITEMS_PATH_ID).handler(this::deleteById);
 
-    router.post(RELATIVE_ITEMS_PATH_ID + "/mark-in-process")
-      .handler(handle(this::markAsInProcess));
-    router.post(RELATIVE_ITEMS_PATH_ID + "/mark-in-process-non-requestable")
-      .handler(handle(this::markAsInProcessNonRequestable));
+    Arrays.stream(ItemStatusName.values())
+      .map(x->ItemStatusURL.getUrlForItemStatusName(x))
+      .filter(x->x.isPresent())
+      .forEach(x->router.post(RELATIVE_ITEMS_PATH_ID+x.get()).handler(handle(this::markItemAsTargetStatus)));
+
+//    router.post(RELATIVE_ITEMS_PATH_ID + "/mark-in-process")
+//      .handler(handle(this::markAs));
+//    router.post(RELATIVE_ITEMS_PATH_ID + "/mark-in-process-non-requestable")
+//      .handler(handle(this::markAs));
```

Comment:
As this code has been replaced by the above, please remove this commented out code. If it is needed in the future, it can be recovered from source control.